docs_src/microservices/application/ContextAPI.md

```diff
@@ -1,108 +1,142 @@
 
-The `LoggingClient` exposed on the context is available to leverage logging libraries/service utilized throughout the EdgeX framework. The SDK has initialized everything so it can be used to log `Trace`, `Debug`, `Warn`, `Info`, and `Error` messages as appropriate. See [simple-filter-xml/main.go](https://github.com/edgexfoundry-holding/app-service-examples/blob/master/app-services/simple-filter-xml/main.go) for an example of how to use the `LoggingClient`.
+### EventClient()
 
-### EventClient 
+Returns an `EventClient` to leverage Core Data's `Event` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/interfaces/event.go) for more details. This client is useful for querying events and is used by the [PushToCore](#pushtocore) convenience API described below.
 
-The `EventClient ` exposed on the context is available to leverage Core Data's `Event` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/coredata/event.go#L35) for more details. This client is useful for querying events and is used by the [PushToCore](#pushtocore) convenience API described below.
+### CommandClient() 
 
-### ValueDescriptorClient
+Returns a `CommandClient`  to leverage Core Command's `Command` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/interfaces/command.go) for more details. Useful for sending commands to devices.
 
-The `ValueDescriptorClient ` exposed on the context is available to leverage Core Data's `ValueDescriptor` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/coredata/value_descriptor.go#L29) for more details. Useful for looking up the value descriptor for a reading received.
+### NotificationClient()
 
-### CommandClient 
+Returns a `NotificationClient` to leverage Support Notifications' `Notifications` API. See [README](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/interfaces/notification.go) for more details. Useful for sending notifications. 
 
-The `CommandClient ` exposed on the context is available to leverage Core Command's `Command` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/command/client.go#L28) for more details. Useful for sending commands to devices.
+### SubscriptionClient()
+Returns a `SubscriptionClient` to leverage Support Notifications' `Subscription` API. Note if Support Notifications is not specified in the Clients configuration, this will return nil.
 
-### NotificationsClient
+### DeviceServiceClient()
+Returns a `DeviceServiceClient` to leverage Core Metadata's `DeviceService` API. Note if Core Metadata is not specified in the Clients configuration, this will return nil.
```

Comment:
```suggestion
Returns a `DeviceServiceClient` to leverage Core Metadata's `DeviceService` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/http/deviceservice.go) for more details. Useful for querying information about Device Services. Note if Core Metadata is not specified in the Clients configuration, this will return nil. 
```